File: codegen/results/v002/lmstudio/qwen3.8-27b-mlx_nothink/20260829_154723/typescript/007_beatles_interview/solution.ts
import * as fs from 'fs';

function parseDate(dateStr: string): Date | null {
  if (!dateStr || dateStr === 'null') return null;
  const [month, day, year] = dateStr.split('/').map(Number);
  if (month === undefined || day === undefined || year === undefined) return null;
  return new Date(year, month - 1, day);
}

function formatDate(date: Date | null): string {
  if (!date) return '';
  const year = date.getFullYear();
  const month = String(date.getMonth() + 1).padStart(2, '0');
  const day = String(date.getDate()).padStart(2, '0');
  return `${year}-${month}-${day}`;
}

function calculateAge(birthday: Date | null, died: Date | null): number {
  if (!birthday) return 0;
  const refDate = died ? died : new Date(2025, 6, 1);
  let age = refDate.getFullYear() - birthday.getFullYear();
  const monthDiff = refDate.getMonth() - birthday.getMonth();
  if (monthDiff < 0 || (monthDiff === 0 && refDate.getDate() < birthday.getDate())) {
    age--;
  }
  return age;
}

function parseName(name: string): { firstName: string; lastName: string } {
  const parts = name.trim().split(/\s+/);
  const firstName = parts[0] || '';
  const lastName = parts.length > 1 ? parts[parts.length - 1] : '';
  return { firstName, lastName };
}

function cleanValue(value: string): string {
  const trimmed = value.trim();
  return trimmed === 'null' || trimmed === '' ? '' : trimmed;
}

interface Relative {
  FirstName: string;
  LastName: string;
  Relationship: string;
}

interface Person {
  FirstName: string;
  LastName: string;
  Birthday: string;
  Age: number;
  Relatives: Relative[];
}

function main() {
  const inputPath = process.argv[2] || 'input/input.csv';
  const content = fs.readFileSync(inputPath, 'utf8');
  const lines = content.split(/\r?\n/).filter((line) => line.trim().length > 0);

  const headers = lines[0].split(',');
  const people: Person[] = [];

  for (let i = 1; i < lines.length; i++) {
    const row = lines[i].split(',');
    if (row.length < headers.length) continue;

    const data: { [key: string]: string } = {};
    headers.forEach((h, idx) => {
      data[h] = row[idx] ?? '';
    });

    const nameParts = parseName(data['Name'] || '');
    const birthday = parseDate(data['Birthday']);
    const died = parseDate(data['Died']);

    const relatives: Relative[] = [];
    const relativeCols: Array<[string, string]> = [
      ['Father', 'Father'],
      ['Mother', 'Mother'],
      ['Brother', 'Brother'],
      ['Sister', 'Sister'],
    ];

    for (const [col, relationship] of relativeCols) {
      const value = cleanValue(data[col] || '');
      if (value) {
        const relParts = parseName(value);
        relatives.push({
          FirstName: relParts.firstName,
          LastName: relParts.lastName,
          Relationship: relationship,
        });
      }
    }

    people.push({
      FirstName: nameParts.firstName,
      LastName: nameParts.lastName,
      Birthday: formatDate(birthday),
      Age: calculateAge(birthday, died),
      Relatives: relatives,
    });
  }

  console.log(JSON.stringify(people, null, 2));
}

main();

File: codegen/results/v002/lmstudio/qwen3.8-27b-mlx_nothink/20260829_154723/python/007_beatles_interview/solution.py
import csv, json, datetime

as_of = datetime.date(2025, 7, 1)

with open('input/input.csv') as f:
    reader = csv.DictReader(f)
    rows = list(reader)

def parse_date(s):
    if not s or s.strip().lower() == 'null':
        return None
    return datetime.datetime.strptime(s.strip(), '%m/%d/%Y').date()

def age_at(as_of, birth):
    return as_of.year - birth.year - ((as_of.month, as_of.day) < (birth.month, birth.day))

def split_name(full_name):
    if not full_name or full_name.strip().lower() == 'null':
        return None, None
    parts = full_name.strip().split()
    if len(parts) == 1:
        return parts[0], ''
    return parts[0], ' '.join(parts[1:])

def rel(full_name, relationship):
    if not full_name or full_name.strip().lower() == 'null':
        return None
    first, last = split_name(full_name)
    return {'FirstName': first, 'LastName': last, 'Relationship': relationship}

out = []
for r in rows:
    full_name = r['Name'].strip()
    parts = full_name.split()
    first_name = parts[0]
    last_name = ' '.join(parts[1:]) if len(parts) > 1 else ''

    birth = parse_date(r['Birthday'])
    age = age_at(as_of, birth)
    birthday_str = birth.strftime('%Y-%m-%d')

    relatives = []
    for key, rel_type in [('Father','Father'),('Mother','Mother'),('Brother','Brother'),('Sister','Sister')]:
        val = r.get(key, '')
        rec = rel(val, rel_type)
        if rec:
            relatives.append(rec)

    out.append({
        'FirstName': first_name,
        'LastName': last_name,
        'Birthday': birthday_str,
        'Age': age,
        'Relatives': relatives
    })

print(json.dumps(out, indent=2))